src/Autofac.Integration.AspNetCore.Multitenant/AutofacMultitenantServiceProviderExtensions.cs

```diff
@@ -45,12 +45,12 @@ public static class AutofacMultitenantServiceProviderExtensions
         /// </exception>.
         public static MultitenantContainer GetAutofacMultitenantRoot(this IServiceProvider serviceProvider)
         {
-            if (!(serviceProvider.GetAutofacRoot() is MultitenantContainer multitenantContainer))
-            {
+            var root = serviceProvider.GetAutofacRoot();
+
+            if (!root.IsRegistered<MultitenantContainer>())
```

Comment:
If the service provider is multitenant, wouldn't the original logic here still work?

Reply:
The service provider is not a MTC anymore. We don't register it as `IServiceProvider` anymore. It's being registered as `MultitenantContainer`.

Reply:
Hang on, though. If the middleware [gets a service provider so it can locate the multitenant container](https://github.com/autofac/Autofac.AspNetCore.Multitenant/blob/41a8f486ce38431954414e670223cfaee89b6a1d/src/Autofac.Integration.AspNetCore.Multitenant/MultitenantRequestServicesMiddleware.cs#L56) but it's not a multitenant service provider, what provider is it?